### PR TITLE
feat: sociable unit test functionality

### PR DESCRIPTION
**Pull request by omermorad**

### CODE REVIEW

**Comment by omermorad** [comment]
@soryy708 up to review?

File: packages/unit/src/main.ts [hunk 1]
```diff
@@ -23,45 +24,58 @@ const SuitesDIAdapters = {
...
+  }
+
+  public static sociable<TClass = any>(targetClass: Type<TClass>): SociableTestBedBuilder<TClass> {
+    return testBedBuilderFactory(SuitesDIAdapters, SuitesDoublesAdapters, targetClass).create(
+      SolitaryTestBedBuilder
+    );
   }
 
-  public static create<TClass = any>(targetClass: Type<TClass>): TestBedBuilder<TClass> {
+  public static create<TClass = any>(targetClass: Type<TClass>): SolitaryTestBedBuilder<TClass> {
```
Comment (soryy708) [inline packages/unit/src/main.ts:78]: Backwards-compatible API. Awesome!

Comment (omermorad) [reply packages/unit/src/main.ts:78]: lol. Yes it'a gonna be a whole new framework under the scope `@suites`.

File: packages/unit/src/main.ts [hunk 2]
```diff
@@ -23,45 +24,58 @@ const SuitesDIAdapters = {
...
 export class TestBed {
-  public static solitary<TClass = any>(targetClass: Type<TClass>): TestBedBuilder<TClass> {
-    return createTestbedBuilder(SuitesDIAdapters, SuitesDoublesAdapters, targetClass);
+  public static solitary<TClass = any>(targetClass: Type<TClass>): SolitaryTestBedBuilder<TClass> {
+    return testBedBuilderFactory(SuitesDIAdapters, SuitesDoublesAdapters, targetClass).create(
+      SolitaryTestBedBuilder
+    );
+  }
+
+  public static sociable<TClass = any>(targetClass: Type<TClass>): SociableTestBedBuilder<TClass> {
```
Comment (soryy708) [inline packages/unit/src/main.ts:72]: I couldn't find any tests covering this new user-facing API?
I see tests that cover the insides, but nothing drives it as a user would?

Comment (omermorad) [reply packages/unit/src/main.ts:72]: There is `e2e` that I need to apply as well, you are right

File: packages/core/__test__/sociable/sociable-testbed-builder.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    unitBuilder = new SociableTestBedBuilder(
+      Promise.resolve(mock),
+      Promise.resolve(FakeAdapter),
+      new UnitMocker(Promise.resolve(mock), Promise.resolve(FakeAdapter)),
+      UserService,
+      console
+    );
+
+    const testBed = await unitBuilder
+      .expose(UserApiService)
```
Comment (soryy708) [inline packages/core/__test__/sociable/sociable-testbed-builder.integration.test.ts:34]: Because this is a public user-facing API, lets discuss the chosen terminology expose, to mark a dependency to not be faked.

What is the mental model Automock tries to give its users about what dependencies get replaced with test-doubles and what aren't?

In my mind for example, since Automock uses the terminology: "Automock, mock...using", I'd expect this to be called dontMock or something like that.

**Comment by soryy708** [comment]
Relates to: https://github.com/automock/automock/issues/230

File: e2e/jest/nestjs/e2e-assets-sociable.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+export class UserService {
+  public constructor(
+    private readonly userApiService: UserApiService,
+    private readonly userDal: UserDal,
+    private readonly logger: Logger,
+    @Inject('SOME_VALUE_TOKEN') private readonly someValue: string[]
+  ) {
+    this.logger.log('Just logging a message');
+
+    if (typeof this.logger.log !== 'function') {
```
Comment (soryy708) [inline e2e/jest/nestjs/e2e-assets-sociable.ts:113]: If `this.logger.log` isn't a function then we'll never reach this check, because of the previous `this.logger.log` invocation (that will throw)

File: packages/unit/src/types.ts [hunk 5]
```diff
@@ -206,7 +204,7 @@ export interface MockOverride<TDependency, TClass> {
   /**
    * Specifies a constant value to be used for the mocked dependency.
    *
-   * @since 2.0.0
```
Comment (soryy708) [inline packages/unit/src/types.ts:209]: why was this changed?

File: packages/core/__test__/sociable/sociable-testbed-builder.integration.test.ts (hunk 3 above)
Comment (omermorad) [reply packages/core/__test__/sociable/sociable-testbed-builder.integration.test.ts:35]: @potoos @qballer what do you think?
